src/BizHawk.Emulation.Cores/Consoles/Nintendo/BSNES/BsnesCore.IMemoryDomains.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using BizHawk.Emulation.Common;
+
+namespace BizHawk.Emulation.Cores.Nintendo.BSNES
+{
+	public partial class BsnesCore
+	{
+		private IMemoryDomains _memoryDomains;
+
+		private unsafe void SetMemoryDomains()
+		{
+			List<MemoryDomain> mm = new();
+			foreach (int i in Enum.GetValues(typeof(BsnesApi.SNES_MEMORY)))
+			{
+				void* data = Api.core.snes_get_memory_region(i, out int size, out int wordSize);
+				if (data == null) continue;
+				if (Enum.GetName(typeof(BsnesApi.SNES_MEMORY), i) == BsnesApi.SNES_MEMORY.CARTRIDGE_RAM.ToString())
+				{
+					_saveRam = (byte*) data;
+					_saveRamSize = size;
+				}
+				mm.Add(new MemoryDomainIntPtr(Enum.GetName(typeof(BsnesApi.SNES_MEMORY), i), MemoryDomain.Endian.Little, (IntPtr) data, size, true, wordSize));
```

Comment:
These raw core pointers are in waterbox space, so you need to Enter / Exit when using the memory domain.  `MemoryDomainIntPtrMonitor` should take care of a lot of the work.